cmp/zoneGrid/ZoneGridModel.ts

```diff
@@ -611,46 +621,77 @@ export class ZoneGridModel extends HoistModel {
+        strict: boolean
     ): Record<Zone, ZoneMapping[]> {
         const ret = {} as Record<Zone, ZoneMapping[]>;
         forOwn(mappings, (rawMapping, zone) => {
             // 1) Standardize mapping into an array of ZoneMappings
-            const mapping = [];
+            let mapping = [];
             castArray(rawMapping).forEach(it => {
                 if (!it) return;
 
                 const ret = isString(it) ? {field: it} : it,
                     col = this.findColumnSpec(ret);
 
-                throwIf(!col, `Column not found for field ${ret.field}`);
-                return mapping.push(ret);
+                if (col) {
+                    mapping.push(ret);
+                } else {
+                    const message = `Column not found for field '${ret.field}'`;
+                    this.throwOrWarn(message, strict);
+                }
             });
 
             // 2) Ensure mapping respects configured limits
             const limit = this.limits?.[zone];
             if (limit) {
-                throwIf(
-                    isFinite(limit.min) && mapping.length < limit.min,
-                    `Requires minimum ${limit.min} mappings in zone "${zone}"`
-                );
-                throwIf(
-                    isFinite(limit.max) && mapping.length > limit.max,
-                    `Exceeds maximum ${limit.max} mappings in zone "${zone}"`
-                );
+                if (isFinite(limit.min) && mapping.length < limit.min) {
+                    const message = `Requires minimum ${limit.min} mappings in zone "${zone}. Setting to default."`;
+
+                    this.throwOrWarn(message, strict);
+
+                    mapping = this._defaultState.mappings[zone];
+                }
+
+                if (isFinite(limit.max) && mapping.length > limit.max) {
+                    const diff = mapping.length - limit.max,
+                        message = `Exceeds maximum ${limit.max} mappings in zone "${zone}". Dropping last ${diff} fields`;
+
+                    this.throwOrWarn(message, strict);
+
+                    mapping = dropRight(mapping, mapping.length - limit.max);
+                }
 
                 if (!isEmpty(limit.only)) {
+                    const offenders = [];
                     mapping.forEach(it => {
-                        throwIf(
-                            !limit.only.includes(it.field),
-                            `Field "${it.field}" not allowed in zone "${zone}"`
-                        );
+                        if (!limit.only.includes(it.field)) {
+                            const message = `Field "${it.field}" not allowed in zone "${zone}. Removing"`;
+
+                            this.throwOrWarn(message, strict);
+
+                            offenders.push(it.field);
+                        }
                     });
+
+                    if (!isEmpty(offenders)) {
+                        remove(mapping, it => offenders.includes(it.field));
```

Comment:
set to default again here. Only need to warn for one field and say we're going to the default immediately at that point.